### DISCUSSION TITLE
Different upstream DNS for different TLD

### DISCUSSION THREAD

**Discussion by ammnt** [AdguardTeam/AdGuardHome#6248] [Q&A] [answered]
Hi,

Please explain how to use different upstream DNS servers for different top level domains (zones). For example I want to setup:
- 1.1.1.1 for all domains and subdomains in .COM, .NET and .ORG zones;
- 8.8.8.8 for all domains and subdomains in .INFO, .US and .EU zones;
- 9.9.9.9 for all domains and subdomains in .PL, .IO and .FR zones;
... etc

Thank you😘

**Comment by ainar-g** [comment]
Have you tried the domain-specific upstreams as [described in the Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains)?

**Reply by ammnt** [reply]
Yeah. I want to add Web3 domains support and added this rule in client's upstream section:
[/eth/]45.90.28.165:53

But ENS domains still didn't work for me:
vitalik.eth
exchange.eth
brave.eth
etc.
![image](https://github.com/AdguardTeam/AdGuardHome/assets/45385632/7b7910dd-28f7-4b66-bc74-ebacc5eb79c6)

Maybe something wrong in queries routing?

I also have no logs in NextDNS UI:
![image](https://github.com/AdguardTeam/AdGuardHome/assets/45385632/b0931294-7064-410f-987d-4efb111de357)

Everything works perfect if I just use NextDNS as upstream DNS🫣 Please check this too!

Thank you.

**Reply by ainar-g** [reply]
Are you sure that the IP address is correct?  Because none of these domains resolve with e.g. `dig`:

```sh
dig IN A 'brave.eth' @45.90.28.165
```

```none
…
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 33749
…
```

**Reply by ammnt** [reply]
@ainar-g, sure. You don't get anything because this server is assigned to my IP🙂

**Reply by ainar-g** [reply, marked as answer]
Well if that upstream is sensitive to the source address, have you checked that your setup is assigning the correct one when querying it?

**Reply by ammnt** [reply]
@ainar-g, you were right. This crap doesn't work anymore for web3 domains....
